dk61_hlovatskyi/lab2_simple_kernel_module/README.rst

```diff
@@ -0,0 +1,85 @@
+===========================
+**Lab2: Simple_kernel_module**
+===========================
+
+**Description**
+
+There we have an instance of kernel module. 
+It has privileges compare to userspace-applications.
+You can't use standard 	libraries, 'cause modules need to be fast and light. 
+It is not safe to insert and remove modules, unless if you know what you are doing. 
+
+
+**Explaining**
+
+After including all linux libraries that we need to use we define module license as "Dual MIT/GPL" this disabling lock debugging due to kernel taint. 
+Execution starts from the function ``module_init(<init_func>)`` and end with ``module_exit(<exit_func>);``
+Result of ``printk()`` is a warning message that notify us about inserting this module to the kernel and print ``jiffies`` - amount of interruptions that happened from the system boot. 
+
+
+The task was to implement tasklet, that print the same variable ``jiffies`` and compare results of execution of this code on x86 and BBXM. Also add opportunity to set value of the username for system's greeting from command line during inserting the module.  
+
+
+Tasklets are a deferral scheme that you can schedule for a registered function to run later. They are atomic that fact exclude using primitives sync and  their priority (bottom-half) is less than ISR(top-half).
+Their structure defined in the <linux/interrupt.h>. 
+It is usefull to use it for    
+
+**Implementing**
+We created tasklet with a static method by MACROS ``DECLARE_TASKLET(name, func, data);`` which is equivalent for the next declaration:
+``struct tasklet_struct my_tasklet = {
+
+ NULL, 0, ATOMIC_INIT(0), tasklet_handler, dev
+
+};``
+We created tasklet called ``my_tasklet``, where attribute ``count = 0``, it means that tasklet has approval for execution. The ``tasklet_handler`` wiil be handler of this tasklet and it receive argument `dev`` as input value. 
+For dynamic initializition use ``struct tasklet_struct* t`` and call it by ``tasklet_init(t, tasklet_handler, dev)``
```

Comment:
use `struct tasklet_struct *tsk = kmalloc(sizeof *tsk, GFP_KERNEL)` and call `tasklet_init` afterwards